src/main/scala/firrtl/Emitter.scala

```diff
@@ -863,13 +863,9 @@ class VerilogEmitter extends SeqTransform with Emitter {
   }
 
   override def execute(state: CircuitState): CircuitState = {
-    val newAnnos = state.annotations.flatMap {
-      case EmitCircuitAnnotation(_) =>
-        val writer = new java.io.StringWriter
-        emit(state, writer)
-        Seq(EmittedVerilogCircuitAnnotation(EmittedVerilogCircuit(state.circuit.main, writer.toString)))
-
-      case EmitAllModulesAnnotation(_) =>
+    val oneFilePerModule = state.annotations.collect{ case a: EmitOneFilePerModuleAnnotation.type => a }.nonEmpty
```

Comment:
collectFirst{...}.isDefined might be faster because it can exit early

Reply:
Makes sense to me. Fix pending.